### PR TITLE
Document persist failure supervision for ESBs

### PR DESCRIPTION
**Pull request by chbatey**
Refs https://github.com/akka/akka/issues/30062

### CODE REVIEW

**Review by johanandren — APPROVED** [review]

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala [hunk 1]
```diff
@@ -31,6 +31,10 @@ abstract class EventSourcedBehavior[Command, Event, State] private[akka] (
   }
 
   /**
+   * If using onPersistFailure the supervision is only around the event sourced behavior not any outer setup/withTimers
+   * block. If using restart any actions e.g. scheduling timers, can be done on the PreRestart signal or on the
```
Comment (adkafka) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala:35]: The `PreRestart` signal says the following:
> Lifecycle signal that is fired upon restart of the Actor before replacing the behavior with the fresh one (i.e. this signal is received within the behavior that failed).

Given that is the case, it seems impossible to use this signal to affect the *new* behavior. Or am I missing something?

**Comment by chbatey** [comment]
The behaviour will be the old one but if you schedule a timer it'll be sent
to the new behavior

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala [hunk 2]
```diff
@@ -208,6 +208,10 @@ object EventSourcedBehavior {
    * Specifically BackOff to prevent resume being used. Resume is not allowed as
    * it will be unknown if the event has been persisted.
    *
+   * This supervision is only around the event sourced behavior not any outer setup/withTimers
+   * block. If using restart any actions e.g. scheduling timers, can be done on the PreRestart
+   * signal or on the RecoveryCompleted signal.
```
Comment (octonato) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala:213]: I don't understand that sentence. Do you mean with "using restart any actions"?

Comment (patriknw) [reply akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala:213]: I read it as "if using restart"  = if the backoff restart supervision is enabled by defining it with this method

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala (hunk 2 above)
Comment (octonato) [reply akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala:213]: In that case, a comma could help. 

```suggestion
   * block. If using restart, any actions e.g. scheduling timers, can be done on the PreRestart
```

**Pull request merged**